Update oneVPL samples

jeffreymcallister opened the pull request:
# Introduction
- Changes to existing samples

# Existing Sample Changes
## Description

This PR includes multiple updates to enable use on GPU as well as to align with oneVPL spec updates.

## External Dependencies

In Linux the sample build searches for standard GPU stack dependencies.  If they are not found the samples gracefully build without GPU support.

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [X ] When compiling the compliler flag "-Wall -Wformat-security -Werror=format-security" was used

JoeOster commented:
@mav-intel - Can you take a look at this as your oneVPL domain expert. With these substantial changes, I just want confirmation from you they are in line with your expectations.

JoeOster reviewed (dismissed):
approved for ci only

JoeOster reviewed (approved):
approved for ci only

tomlenth commented on Libraries/oneVPL/dpcpp-blur/README.md:
```diff
@@ -32,8 +33,11 @@ If the oneAPI DPC++ Compiler is not found, the blur operation will be disabled.
+| Output resolution | 256x192
 
+## Linux GPU optimizations
+
```
Can you explain what this does a little.  it's very out of context.  If you intend for the user to enter this command, it should be in the same code format as the commands for running the sample.

jeffreymcallister replied:
Thanks for finding this -- this section was not for this version of the sample.  Removed.  For general GPU setup the user can follow the oneAPI basekit setup instructions.

tomlenth reviewed (changes requested):
Please include in the readmes the sections specific to the include file and running in the devcloud.  Examples can be found in the mandelbort readme: https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot

mliu2020 commented on Libraries/oneVPL/dpcpp-blur/README.md:
````diff
@@ -92,10 +93,10 @@ Perform the following steps:
    ```
    Here `<oneapi_install_dir>` represents the root folder of your oneAPI
-   installation, which is `C:\Program Files (x86)\Intel\oneAPI\`
+   installation, which is which is `C:\Program Files (x86)\Intel\oneAPI\`
````
Does this double "which is"?

Mark

mliu2020 commented on Libraries/oneVPL/hello-decode/README.md:
````diff
@@ -87,10 +88,10 @@ Perform the following steps:
    ```
    Here `<oneapi_install_dir>` represents the root folder of your oneAPI
-   installation, which is `C:\Program Files (x86)\Intel\oneAPI\`
+   installation, which is which is `C:\Program Files (x86)\Intel\oneAPI\`
````
"which is" is doubled.

mliu2020 commented on Libraries/oneVPL/hello-vpp/README.md:
````diff
@@ -85,10 +85,10 @@ Perform the following steps:
    ```
    Here `<oneapi_install_dir>` represents the root folder of your oneAPI
-   installation, which is `C:\Program Files (x86)\Intel\oneAPI\`
+   installation, which is which is `C:\Program Files (x86)\Intel\oneAPI\`
````
"which is" is doubled.

JoeOster commented on Libraries/oneVPL/dpcpp-blur/README.md:
```diff
@@ -1,38 +1,36 @@
 # `dpcpp-blur` Sample
```
I made a few grammar and punctuation changes to this document.

JoeOster commented on Libraries/oneVPL/hello-decode/README.md:
```diff
@@ -1,11 +1,11 @@
 | Optimized for    | Description
 |----------------- | ----------------------------------------
-| OS               | Ubuntu* 18.04; Windows* 10
+| OS               | Ubuntu* 20.04; Windows* 10
```
Most samples work in 18.04 and 20.04, is there a reason for the change?

jeffreymcallister replied:
Only 20.04 is supported for GPU.

JoeOster commented on Libraries/oneVPL/hello-decode/README.md:
```diff
@@ -1,11 +1,11 @@
 # `hello-decode` Sample
```
Please note I made some grammar and punctuation changes

JoeOster reviewed (changes requested):
I altered all the readmes with minor changes, please alter if I changed the messaging to much and or let me know if you are good with the changes

mav-intel commented:
> @mav-intel - Can you take a look at this as your oneVPL domain expert. With these substantial changes, I just want confirmation from you they are in line with your expectations.

Hi Joe. Yes these changes are POR and expected

jeffreymcallister commented:
Accepted all changes, added additional readme cleanup, and added sections requested by Tom.  Squashed so that everything is ready to go in as one commit.

JoeOster reviewed (approved):
approved for ci

JoeOster reviewed (approved):
approved for ci
